### PR TITLE
:bug:Source Close.com: Fix Email Activities Stream Pagination

### LINKED ISSUE #20959: Source Close.com - connector fails to synchronize large number of items
## Environment
- **Airbyte version**:  irrelevant
- **OS Version / Instance**: irrelevant
- **Deployment**: irrelevant
- **Source Connector and version**: [source-close-com](https://docs.airbyte.io/integrations/sources/close-com) 0.2.0
- **Destination Connector and version**: irrelevant
- **Step where error happened**: Initial Sync

## Current Behavior
It is not possible to synchronize the whole history of the Email Activities stream because Airbyte is making use of the `_skip` parameter of the API to paginate the results but it has an upper limit of 25000. So when the total number of items exceeds this number, the sync fails with below error :
````
"message":"Something went wrong in the connector. See the logs for more details.","internal_message":"Request to https://api.close.com/api/v1/activity/email/?_skip=250100&_limit=100&date_created__gt=2021-03-01T00%3A00%3A00Z&date_created__lte=2022-02-28T00%3A00%3A00Z failed with status code 400 and error message **The skip you set is larger than the maximum skip for this resource (max_skip = 250000)**.
Please refine your search query.","stack_trace":"Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 13, in <module>
    launch(source, sys.argv[1:])
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 131, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 122, in run
    for message in generator:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/manifest_declarative_source.py", line 112, in read
    yield from super().read(logger, config, catalog, state)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 125, in read
    raise e
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 111, in read
    yield from self._read_stream(
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 180, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 243, in _read_incremental
    for message_counter, record_data_or_message in enumerate(records, start=1):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/declarative_stream.py", line 115, in read_records
    for record in self.retriever.read_records(sync_mode, cursor_field, stream_slice, stream_state):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 374, in read_records
    for record in records_generator:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py", line 430, in _read_pages
    yield from records_generator_fn(request, response, stream_state, stream_slice)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 415, in parse_records_and_emit_request_and_responses
    yield from self.parse_response(response, stream_slice=stream_slice, stream_state=stream_state)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 326, in parse_response
    raise ReadException(error_message)
````

## Expected Behavior
To iterate through a long list of results Close.com recommends to update the query so that fewer results are returned, for instance using the `date_created` field. Alternatively the Export API can be used.
It is fully explained here: https://developer.close.com/topics/pagination/#deep-pagination

## Are you willing to submit a PR?
Not at the moment

### PR DESCRIPTION
**Pull request by topefolorunso**
## What
Fixes #20959

## How
The stream previously used _skip parameter for pagination. That was what caused the error. According to the API doc, in such cases, it is better to use the date_created field for this. But in airbyte, the date_created field is best suited for incremental sync, so the pagination is handled through incremental sync. Makes sense?

Please see here for reference - https://developer.close.com/topics/pagination/#deep-pagination

## Recommended reading order
1. `manifest.yaml`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Sorry the long delay to get into your contribution. Probably this week or next one the team will start the process. Are you able to run the tests in your local setup? It can speed up a lot the process if you share the integration tests results

File: airbyte-integrations/connectors/source-close-com/source_close_com/manifest.yaml [hunk 1]
```diff
@@ -194,9 +194,10 @@ definitions:
       name: "email_thread_activities"
       path: "activity/emailthread"
   email_activities_stream:
-    $ref: "#/definitions/activities_base_stream"
+    $ref: "#/definitions/base_stream"
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-close-com/source_close_com/manifest.yaml:197]: how disabled pagination can help with limiting error?

Comment (topefolorunso) [reply airbyte-integrations/connectors/source-close-com/source_close_com/manifest.yaml:197]: @lazebnyi The stream previously used `_skip` parameter for pagination. That was what caused the error. According to the API doc, in such cases, it is better to use the `date_created` field for this. But in airbyte, the `date_created` field is best suited for incremental sync, so the pagination is handled through incremental sync. Makes sense?

Please see here for reference - https://developer.close.com/topics/pagination/#deep-pagination

**Comment by jpandl19** [comment]
@marcosmarxm @lazebnyi are you still targeting this week for this one to go live? 

Thanks in advance! (and thank you @topefolorunso for the fix - this one has been blocking my team for months)

File: airbyte-integrations/connectors/source-close-com/source_close_com/manifest.yaml (hunk 1 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-close-com/source_close_com/manifest.yaml:197]: please add this to PR description in How section

**Review by lazebnyi — APPROVED** [review]
lgtm

**Review by marcosmarxm — APPROVED** [review]

**Comment by anatolec** [comment]
Hi,
I am not sure what to do after this fix. Even if I use an Incremental sync, the initial sync would need to throw several requests with different `date_created` values to circumvent the `_skip` limitation. How to set this up ?

**Pull request merged**